From 543187d001f38bd0bbef39675b3f88e41498d857 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Thu, 17 Jul 2025 11:22:03 +0800
Subject: [PATCH 217/236] mtk: hostapd: increase csa count for radar triggered
 channel switch

Increase the CSA count for radar-triggered channel switches.
This allows for more processing time in the case of mbss.
Additionally, it is recommenanded to set default csa count > 6, as the WFA test
plan (ac 4.2.16.1) requires at least 6 beacons to contain CSA IE.

Set all the default csa count to 10

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 src/ap/dfs.c           | 6 +++---
 src/ap/drv_callbacks.c | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/ap/dfs.c b/src/ap/dfs.c
index fa3af65c4..4af12b2b3 100644
--- a/src/ap/dfs.c
+++ b/src/ap/dfs.c
@@ -1089,7 +1089,7 @@ static int hostapd_dfs_request_channel_switch(struct hostapd_iface *iface,
 
 	/* Setup CSA request */
 	os_memset(&csa_settings, 0, sizeof(csa_settings));
-	csa_settings.cs_count = 5;
+	csa_settings.cs_count = 10;
 	csa_settings.block_tx = 1;
 	csa_settings.link_id = -1;
 #ifdef CONFIG_IEEE80211BE
@@ -2035,8 +2035,8 @@ int hostapd_dfs_handle_csa(struct hostapd_iface *iface,
 	bool update_background = false;
 
 	if (!dfs_use_radar_background(iface)) {
-		if (settings->cs_count < 5)
-			settings->cs_count = 5;
+		if (settings->cs_count < 10)
+			settings->cs_count = 10;
 		settings->block_tx = cac_required;
 		return 0;
 	}
diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index b7234f1d3..20205e730 100644
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -2760,7 +2760,7 @@ static void hostapd_event_pp_bitmap_update(struct hostapd_data *hapd,
 
 	/* Setup CSA request */
 	os_memset(&csa_settings, 0, sizeof(csa_settings));
-	csa_settings.cs_count = 5;
+	csa_settings.cs_count = 10;
 	csa_settings.block_tx = 0;
 	csa_settings.link_id = ch_switch->link_id;
 
-- 
2.45.2

